beamer/agent/config.py

```diff
@@ -110,9 +113,13 @@ def load(config_path: Path, options: dict[str, Any]) -> Config:
         raise ConfigError(f"missing settings: {missing}")
 
     rpc_urls = {}
+    chain_confirmation_blocks = {}
 
     for chain_name, chain_info in config["chains"].items():
         rpc_urls[chain_name] = URL(chain_info["rpc-url"])
+        chain_confirmation_blocks[chain_name] = int(
```

Comment:
general suggestion: Wouldn't it be nicer to have a "chain" dictionary in the config which stores chain related information. 
Since we sort our information per chain we could have a dataclass `ChainConfig`
and a `dict[chain_name, ChainConfig]` (kind of similar how we structured it with api schema in the deployment config

Reply:
i can handle this after merging this pr

Reply:
> and a `dict[chain_name, ChainConfig]` (kind of similar how we structured it with api schema in the deployment config

Exactly this has been my plan to do, just didn't want to block this PR on that. But yes, using the apischema approach for agent config is definitely the way to go.

Reply:
ok then lets go for it after this PR, can you create a follow up please?